src/libmongoc/tests/test-mongoc-primary-stepdown.c

```diff
@@ -241,8 +241,7 @@ test_not_primary_keep_pool (mongoc_client_t *client)
       coll, tmp_bson ("{'test': 1}"), NULL, NULL, &error);
    ASSERT (!res);
    ASSERT_CMPINT (error.code, ==, 10107);
-   ASSERT_CONTAINS (error.message,
-                    "Failing command due to 'failCommand' failpoint");
+   ASSERT_CONTAINS (error.message, "failpoint");
```

Comment:
SERVER-57040 changed the error message returned from errors triggered by failpoints.

Reply:
General question: are the changes usually slow enough that all drivers and their tests can just play follow-up? Has there been thought put toward storing them in a shared resource file (or some other strategy to centralize them)? Is that valuable?

Reply:
> are the changes usually slow enough that all drivers and their tests can just play follow-up?

In practice, it is rare that an error message change occurs and requires an update to tests. If I had to guess, once every few months.

> Has there been thought put toward storing them in a shared resource file (or some other strategy to centralize them)? Is that valuable?

That is an interesting idea. I do not know of any effort to centralize error messages. IMO drivers should prefer not checking error messages. There is no guarantee they remain stable server side. I believe the order of preference for driver error handling is:

1. Check the reply `errorLabels`.
2. Check the reply `code`.
3. Check the reply `message`.

An example of (1) being preferred to (2) is [retryable writes error handling](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#retryablewriteerror-labels). Error labels convey the intent of handling from the server. It gives the server the flexibility to add / remove errors that are considered retryable. In pre 4.4 servers, there are no `errorLabels`, so there is a fallback list of error codes.

An example of (2) being preferred to (3) is the [SDAM](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#not-writable-primary-and-node-is-recovering) error handling. It checks for an error code first, but falls back to error message checking.